### ISSUE TITLE
GHCJS foreign import support

### ISSUE THREAD

**Issue by geraldus** [haskell-suite/haskell-src-exts#232]
Hi folks!

I want `stylish-haskell` to support GHCJS' foreign import declarations, Adam Bergmark said that parsing related stuff should be implemented in this package.  I'm just an amateur haskeller rookie, but if someone can give a bit of guidance I will be happy to give a try and provide a PR for this.

GHCJS foreign imports are look like this:

```
foreign import javascript unsafe "document.getElementById($1)"
  js_documentElementById :: JSString -> IO Elem
```

**Comment by mpickering** [comment]
The parser can be found in `InternalParser.ly`. It is basically a simple copy of the GHC parser so it would be best to look at how it is parsed in the GHC parser and then extend the HSE representation with appropriate modifications.

**Comment by geraldus** [comment]
@mpickering hi! Looks like this will be easier than I thought, likely all I need is to add missing `javascript` calling convention, because GHCJS' foreign imports very similar to `ccall`s for example.  Am I right?  Also, how can test my changes?

**Comment by geraldus** [comment]
Ok, I've added missing `javascript` keyword, bumped version number and build `stylish-haskell` with new version of HSE.  But when I tried it on one of my GHCJS files, I still have parse error:

> Language.Haskell.Stylish.Parse.parseModule: could not parse src/Internal/FFI.hs: ParseFailed (SrcLoc "<unknown>.hs" 11 16) "Parse error: javascript"

I've tried to enable `ForeignFunctionInterface` and `JavaScriptFFI` extensions, but this changes nothing. What am I miss?

Here is my changes https://github.com/geraldus/haskell-src-exts/commit/82fa8a9108ab61355a7a399a5d688f134608f611

Also instead of filename it shows `<unknown>`, is it expected output?
